modules/prompters.py

```diff
@@ -138,7 +138,11 @@ def _get_trimmed_message(node: Node, token_usage_fraction: float) -> Message:
         return message
 
     content = notice_retroactively_using_saved_output.format(
-        filename=node.metadata["saved_output_filename"],
+        saved_output_instructions=(
+            f"The full output is saved as {saved_output_filename}. "
+            if (saved_output_filename := node.metadata.get("saved_output_filename"))
+            else ""
+        ),
```

Comment:
This was the original bug

Reply:
Thanks for looking into this! 
I am confused however: why are there sometimes truncated outputs, but no saved_output_filenames?

Reply:
`saved_output_filenames` and truncated outputs are separate things. This code here does the truncation, whereas saving output to a file is done elsewhere. This code is currently assuming that outputs of commands are always saved to a file, which is not true (it depends on the actor).